### PR TITLE
fix: dynamic links for reports

### PR DESCRIPTION
**Pull request by pateljannat**
**Issue:**

1. The issue was present in three reports.

- Bank Clearance Summary
- Prospect Engaged But Not Converted.
- Payment Period Based on Invoice Date

2. The Dynamic transaction Links were not opening the correct document. 
3. Instead an error used to appear.

Screenshot 2021-01-25 at 5 05 58 PM

### CODE REVIEW

File: erpnext/crm/report/prospects_engaged_but_not_converted/prospects_engaged_but_not_converted.py [hunk 1]
```diff
@@ -19,15 +19,48 @@ def set_defaut_value_for_filters(filters):
...
+		},
+		{
+			"label": _("Organization"),
+			"fieldname": "organization",
+			"width": 120
+		},
+		{
+			"label": _("Reference Document"),
+			"fieldname": "reference_document",
+			"width": 100
```
Comment (hasnain2808) [inline erpnext/crm/report/prospects_engaged_but_not_converted/prospects_engaged_but_not_converted.py:44]:
```	
"fieldtype": "Link",
"options": "Doctype",
```

File: erpnext/accounts/report/bank_clearance_summary/bank_clearance_summary.py [hunk 2]
```diff
@@ -15,15 +15,49 @@ def execute(filters=None):
...
-		_("Posting Date") + ":Date:100",
-		_("Cheque/Reference No") + "::120",
-		_("Clearance Date") + ":Date:100",
-		_("Against Account") + ":Link/Account:170",
-		_("Amount") + ":Currency:120"
-	]
+	columns = [{
+			"label": _("Payment Document"),
+			"fieldname": "payment_document",
+			"width": 130
```
Comment (hasnain2808) [inline erpnext/accounts/report/bank_clearance_summary/bank_clearance_summary.py:23]:
```	
"fieldtype": "Link",
"options": "Doctype",
```

File: erpnext/accounts/report/bank_clearance_summary/bank_clearance_summary.py [hunk 3]
```diff
@@ -15,15 +15,49 @@ def execute(filters=None):
...
-		_("Payment Entry") + ":Dynamic Link/"+_("Payment Document")+":110",
-		_("Posting Date") + ":Date:100",
-		_("Cheque/Reference No") + "::120",
-		_("Clearance Date") + ":Date:100",
-		_("Against Account") + ":Link/Account:170",
-		_("Amount") + ":Currency:120"
-	]
+	columns = [{
+			"label": _("Payment Document"),
+			"fieldname": "payment_document",
```
Comment (hasnain2808) [inline erpnext/accounts/report/bank_clearance_summary/bank_clearance_summary.py]: "label": _("Payment Document Type"),
"fieldname": "payment_document_type",

File: erpnext/accounts/report/bank_clearance_summary/bank_clearance_summary.py [hunk 4]
```diff
@@ -15,15 +15,49 @@ def execute(filters=None):
...
+			"width": 130
+		},
+		{
+			"label": _("Payment Entry"),
+			"fieldname": "payment_entry",
+			"fieldtype": "Dynamic Link",
+			"options": "payment_document",
```
Comment (hasnain2808) [inline erpnext/accounts/report/bank_clearance_summary/bank_clearance_summary.py]: "options": "payment_document_type",

File: erpnext/crm/report/prospects_engaged_but_not_converted/prospects_engaged_but_not_converted.py [hunk 5]
```diff
@@ -19,15 +19,48 @@ def set_defaut_value_for_filters(filters):
...
+			"width": 120
+		},
+		{
+			"label": _("Organization"),
+			"fieldname": "organization",
+			"width": 120
+		},
+		{
+			"label": _("Reference Document"),
+			"fieldname": "reference_document",
```
Comment (hasnain2808) [inline erpnext/crm/report/prospects_engaged_but_not_converted/prospects_engaged_but_not_converted.py]: "label": _("Reference Document Type"),
"fieldname": "reference_document_type",

File: erpnext/crm/report/prospects_engaged_but_not_converted/prospects_engaged_but_not_converted.py [hunk 6]
```diff
@@ -19,15 +19,48 @@ def set_defaut_value_for_filters(filters):
...
+			"width": 100
+		},
+		{
+			"label": _("Reference Name"),
+			"fieldname": "reference_name",
+			"fieldtype": "Dynamic Link",
+			"options": "reference_document",
```
Comment (hasnain2808) [inline erpnext/crm/report/prospects_engaged_but_not_converted/prospects_engaged_but_not_converted.py]: "options": "reference_document_type",

**Review by hasnain2808 — CHANGES_REQUESTED** [review]

**Comment by hasnain2808** [comment]
👍

**Review by nabinhait — APPROVED** [review]

**Pull request merged**